code/game/machinery/computer/atmos_control.dm

```diff
@@ -467,6 +486,28 @@ var/global/list/atmos_controllers = list()
 					selected_preset.scrubbers_gases[gas] = !selected_preset.scrubbers_gases[gas] //toggle scrubbing for it
 				return 1
 
+	if(href_list["set_mass_mode"])
+		if(!(log_in_id && (access_ce in log_in_id.GetAccess()) || emagged || usr.hasFullAccess()))
+			return 1
+
+		var/list/scrubmodes = list(
+		"SCRUBBING MODE" = AALARM_MODE_SCRUBBING,
+		"REPLACEMENT MODE" = AALARM_MODE_REPLACEMENT,
+		"PANIC SIPHON MODE" = AALARM_MODE_PANIC,
+		"AIR CYCLE MODE" = AALARM_MODE_CYCLE,
+		"FILL MODE" = AALARM_MODE_FILL,
+		"OFF MODE" = AALARM_MODE_OFF
+		)
+		var/choice = input("Select the scrubber-mode you wish to set all air alarms to.\n\nAll air alarms will be set to this mode.", "Select mode", null) as null|anything in scrubmodes
+		if(choice)
+			var/newmode = scrubmodes[choice]
+			var/areyouforreal = alert(usr, "Are you sure you wish to change the mode of every air alarm to [choice]?",,"Yes", "No") == "Yes" ? 1 : 0
```

Comment:
`input()` and `alert()` will `sleep` internally until the user makes a choice, which means you need to re-check that the conditions that allowed them to get to this point are still holding. For instance, the user could have moved away, or died in the meantime. Ideally this would be handled within the nanoui part, so that there's no sleeping involved and no extra checks necessary.

Reply:
i hate UI so much that i will just add more checks